Resources/templates/Doctrine/ListBuilderAction.php.twig

```diff
@@ -112,9 +111,20 @@
 {% endblock %}
 
 {% block addJoinFor -%}
-    protected function addJoinFor($table, $query)
+    protected function addJoinFor($table, $query, $currentObject = true)
     {
-        $query->leftJoin('q.'.$table, $table);
+        if ($currentObject) {
+            $query->leftJoin('q.'.$table, $table);
+        } else {
+            $tables = explode('.', $table);
+            $previousTable = 'q';
+            foreach (array_slice($tables, 0, -1) as $table) {
+                $query->leftJoin(implode('.', array($previousTable, $table)), $table);
```

Comment:
Just wondering, shouldn't we prefix the alias with a `sort_`string to avoid any conflict with user query?